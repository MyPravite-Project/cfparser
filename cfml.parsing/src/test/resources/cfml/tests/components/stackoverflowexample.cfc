component{
	function processLogin(){
		p = '';
		location(x=true, y=false);
	}
	
	
}