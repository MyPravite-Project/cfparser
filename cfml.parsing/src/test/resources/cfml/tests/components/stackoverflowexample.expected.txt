/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
Hidden:NEWLINE                  <>
FUNCTION                 <function>
IDENTIFIER               <processLogin>
'('                      <(>
')'                      <)>
'{'                      <{>
Hidden:NEWLINE                  <>
IDENTIFIER               <p>
'='                      <=>
OPEN_STRING              <'>
CLOSE_STRING             <'>
';'                      <;>
Hidden:NEWLINE                  <>
LOCATION                 <location>
'('                      <(>
IDENTIFIER               <x>
'='                      <=>
BOOLEAN_LITERAL          <true>
','                      <,>
IDENTIFIER               <y>
'='                      <=>
BOOLEAN_LITERAL          <false>
')'                      <)>
';'                      <;>
Hidden:NEWLINE                  <>
'}'                      <}>
Hidden:NEWLINE                  <>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier processLogin)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (assignmentExpression
                (startExpression
                  (baseExpression (unaryExpression (memberExpression (identifier p))))
                )
                =
                (startExpression
                  (baseExpression
                    (unaryExpression (primaryExpression (literalExpression (stringLiteral ' '))))
                  )
                )
              )
              (endOfStatement ;)
            )
            (statement
              (tagOperatorStatement
                (tagFunctionStatement
                  (cfmlFunction location)
                  (
                  (parameterList
                    (parameter
                      (identifier x)
                      =
                      (startExpression
                        (baseExpression (unaryExpression (primaryExpression (literalExpression true))))
                      )
                    )
                    ,
                    (parameter
                      (identifier y)
                      =
                      (startExpression
                        (baseExpression
                          (unaryExpression (primaryExpression (literalExpression false)))
                        )
                      )
                    )
                  )
                  )
                  ;
                )
              )
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public function processLogin()     {
p = '';
location(x = true, y = false);

    }
}
/*======*/